src/common/container.h

```diff
@@ -93,6 +93,24 @@ static inline void smartlist_swap(smartlist_t *sl, int idx1, int idx2)
   }
 }
 
+/** Return true if there is shallow equality between smartlists -
```

Comment:
Perhaps we should put this in `container.c`? That's where the other `_eq()` functions are. I'm not sure if I like functions in .h files, but it seems to be the case in `container.h`, so if you like it like this feel free to leave it as is.

Reply:
Moved to container.c in cda8adc876b87443859577e54f080f56b8f5e583.

Reply:
LGTM